includes/class-customizer.php

```diff
@@ -0,0 +1,218 @@
+	 *
+	 * @param \WP_Customize_Manager $wp_customize The Customizer.
+	 */
+	function customize_register( $wp_customize ) {
+
+		ob_start();
+		esc_html_e( 'Simple Custom CSS allows you to add your own styles or override the default CSS of a plugin or theme.', 'simple-custom-css' );
+		?>
+		<br />
+		<br />
+		<a href="<?php echo esc_url( admin_url( 'themes.php?page=simple-custom-css.php' ) ); ?>"><?php esc_html_e( 'View the full CSS editor', 'simple-custom-css' ); ?>&rarr;</a>
+		<?php
+
+		$description = ob_get_clean();
+
+		$wp_customize->add_section( self::SECTION_NAME, array(
+			'title'       => __( 'Simple Custom CSS', 'simple-custom-css' ),
+			'description' => $description,
+			'capability'  => apply_filters( 'sccss_capability', 'edit_theme_options' ),
+			'priority'    => 101,
+		) );
+
+		$wp_customize->add_setting( $this->setting_name, array(
+				'default'   => '',
+				'type'      => 'option',
+				'transport' => 'postMessage',
+			)
+		);
+
+		// Add control using our custom controller class.
+		$wp_customize->add_control( new Textarea( $wp_customize, 'sccss_textarea', array(
+				'label'       => __( 'Custom CSS', 'simple-custom-css' ),
+				'section'     => self::SECTION_NAME,
+				'settings'    => $this->setting_name,
+				'type'        => 'textarea',
+			)
+		) );
+
+		if ( $wp_customize->is_preview() && ! is_admin() ) {
+			add_action( 'wp_footer', array( $this, 'customize_preview' ), 99 );
+		}
+	}
+
+	/**
+	 * Output the CSS into the Preview pane.
+	 *
+	 * @action wp_footer
+	 *
+	 * @since 3.5
+	 */
+	function customize_preview() {
+		?>
+		<script type="text/javascript">
+			( function( $ ) {
+				wp.customize( <?php echo wp_json_encode( $this->setting_name ); ?>, function( value ) {
+					var style = $( '#sccss-preview' );
+					value.bind(function(to) {
+						style.html( to ? to : '' );
+					});
+				});
+			} )( jQuery )
+		</script>
+		<?php
+	}
+
+	/**
+	 * Enqueue Scripts
+	 *
+	 * @since 3.5
+	 */
+	function customizer_scripts() {
+		$codemirror_css = 'sccss-codemirror-css';
+		$codemirror_js = 'sccss-codemirror-js';
+		$customizer_js = 'sccss-customizer-js';
+		$js_url = trailingslashit( plugins_url( 'simple-custom-css/js/' ) );
+
+		wp_register_style( $codemirror_css, $js_url . 'codemirror/codemirror.css' );
+		wp_enqueue_style( $codemirror_css );
+
+		wp_register_script( $codemirror_js, $js_url . 'codemirror/codemirror-css.min.js', array(), '20160806', true );
+		wp_enqueue_script( $codemirror_js );
+
+		wp_register_script( $customizer_js, $js_url . 'customizer.js', array( 'jquery', $codemirror_js, 'customize-controls' ) );
+
+		$exports = array(
+			'section' => self::SECTION_NAME,
+			'control' => 'sccss_textarea',
+			'element' => $this->setting_name,
+		);
+
+		 wp_scripts()->add_data(
+			 $customizer_js,
+			 'data',
+			 sprintf( 'var _simpleCustomCSSCustomizerExports = %s;', wp_json_encode( $exports ) )
+		 );
+
+		 wp_add_inline_script( $customizer_js, 'simpleCustomCSSCustomizer.init();', 'after' );
```

Comment:
You could eliminate the extra global `_simpleCustomCSSCustomizerExports` and instead do this:

```php
wp_add_inline_script( $customizer_js, sprintf( 'simpleCustomCSSCustomizer.init( %s );', wp_json_encode( $exports ) ), 'after' );
```

Then the `init` function can look at the argument instead of looking at the global. This is better for testing and encapsulation.